sdv/metadata/multi_table.py

```diff
@@ -13,6 +15,165 @@ def __init__(self):
         self._tables = {}
         self._relationships = []
 
+    @staticmethod
+    def _validate_missing_relationship_keys(parent_table, parent_table_name, parent_primary_key,
+                                            child_table_name, child_foreign_key):
+        missing_keys = set()
+        parent_primary_key = cast_to_iterable(parent_primary_key)
+        table_primary_keys = set(cast_to_iterable(parent_table._primary_key))
+        for key in parent_primary_key:
+            if key not in table_primary_keys:
+                missing_keys.add(key)
+
+        if missing_keys:
+            raise ValueError(
+                f'Relationship between tables ({parent_table_name}, {child_table_name}) contains '
+                f'an unknown primary key {missing_keys}.'
+            )
+
+        for key in set(cast_to_iterable(child_foreign_key)):
+            if key not in parent_table._columns:
+                missing_keys.add(key)
+
+        if missing_keys:
+            raise ValueError(
+                f'Relationship between tables ({parent_table_name}, {child_table_name}) '
+                f'contains an unknown foreign key {missing_keys}.'
+            )
+
+    @staticmethod
+    def _validate_missing_relationship_tables(parent_table_name, child_table_name, tables):
+        missing_table_name = set([parent_table_name, child_table_name]) - set(tables)
```

Comment:
What happens if `parent_table_name` or `child_table_name` are None? If it automatically crashes somewhere this is fine, otherwise you have to remove it:

```python3
set([parent_table_name, child_table_name]) - set(tables) - {None}
```

Reply:
No need to remove `None` since if they provide `None` it should crash. And if it does here it's fine too.
```
bool(set([None])) -> True
```